src/gurt/telemetry.c

```diff
@@ -1398,9 +1444,11 @@ d_tm_decrement_gauge(struct d_tm_node_t **metric, uint64_t value,
 	}
 
 	if (node->dtn_type == D_TM_GAUGE) {
-		D_MUTEX_LOCK(&node->dtn_lock);
+		if (node->dtn_protect)
+			D_MUTEX_LOCK(&node->dtn_lock);
```

Comment:
Suggestion if you want to save some lines: what about functions called d_tm_node_lock and d_tm_node_unlock to encapsulate this?